### ISSUE TITLE
Static site cannot be generated

### ISSUE THREAD

**Issue by kbulte** [vuestorefront/vue-storefront#4532]
## Current behavior
I am using vuestorefront version 1.12.0 with capybara theme. I am able to run the site via the dev script. However I wanted to generate vuestorefront as a static site so I can host it via a CDN. When I execute the following commands:

`yarn build`

`yarn generate all`

I get the following typescript errors:

![image](https://user-images.githubusercontent.com/1887008/84317559-4247ad00-ab6d-11ea-84a6-059c261bc52b.png)

## Expected behavior
A folder with the name 'static' is generated and this folder contains the rendered index.html and product pages.

## Steps to reproduce the issue
1. Clone Vuestorefront 1.12.0
1. Install the Capybara theme
1. build the project by executing yarn build
1. generate a static version by executing yarn generate all

## Repository
https://github.com/DivanteLtd/vue-storefront/blob/master/core/scripts/generate.ts

## Environment details
VS code: 1.46.0
OS: macOS catalina 10.15.5
Node: 14.2.9
Yarn: 1.22.4
Typescript: 3.9.4
Vuestorefront: 1.12.0
Capybara theme: 1.0.1

**Comment by medteck** [comment]
This is issue is being very critical for our project. We've been trying to fix it by ourselves (and make a PR) since last week but still no result.
Is there a due date for 1.12.3 ? That would help us figure out if we can wait for that version or not.
Thanks!

**Comment by gibkigonzo** [comment]
I've took a deeper look into this issue and imo it's not gonna be a quick fix. Code is a little bit outdated. Also I think we can do it much simpler and more extendable. Right now there are predefined queries which may look different for each project. So there is no way to add your custom routes.

My proposition for `yarn generate`:
- run normal express app - `yarn start`
- add listener for `afterOutputRenderedResponse` in server.ts to catch output and save it in file
- add custom generate.ts file in theme folder. In this file you will need to export function that has searchAdapter (to fetch data) and routes array (to have a list of routes that we want to render) as arguments. This function will need to return array of string which will be an array of urls that we need to render.

With this approach we will not need to manage separate rendering script (it will be same as server.ts) also we will be sure that it runs same as normal mode. Devs will also have more flexibility with defining what urls they want to render. We can also add info in docs how can may look example `generate.ts` in theme folder.
@Fifciu

**Comment by elvinristi** [comment]
Got same error with `yarn dev`, but I don't think it's the case of VSF. It's like some hidden error which is not shown and therefor doesn't initialise during build some parts.

I had custom module which included `src/modules/<module-name>/server.ts`. When I removed the file then there was no error anymore.

After putting it back I debugged it to the line where was used `getApiEndpointUrl()` - have custom server-side proxy on POST (payment gateway callback).

After replacing this usage I got rid of this issue.

In my case>
```diff
- import getApiEndpointUrl from '@vue-storefront/core/helpers/getApiEndpointUrl';
...
...
-      const apiUrl = getApiEndpointUrl(configMakecommerce.endpoint.callbacks, callType);
+      const apiUrl = configMakecommerce.endpoint.callbacks[callType];
```
